lib/cocoapods/target/build_settings.rb

```diff
@@ -966,6 +998,14 @@ def pod_target_xcconfig_values_by_consumer_by_key
           end
         end
 
+        # @return [Boolean]
+        define_build_settings_method :should_apply_xctunwrap_fix?, :memoized => true do
+          library_xcconfig? &&
+            target.platform.name == :ios &&
+            Version.new(target.platform.deployment_target) < Version.new('12.2') &&
+            target.library_spec_consumers.flat_map(&:frameworks).include?('XCTest')
```

Comment:
should this be checking the `frameworks` method in build settings rather than going into the spec consumers?

Reply:
actually I avoided that because `:frameworks`build settings method includes those of test specs and app specs. In my sample this has not been an issue if the test bundle itself brings in XCTest but a library we depend on doesn't.

Reply:
> actually I avoided that because :frameworksbuild settings method includes those of test specs and app specs

but in this case, you already have a conditional check for `library_xcconfig?` in this `&&` chain

Reply:
If a pod target has 2 specs, 1 library spec and 1 test spec and the test spec defines `test_spec.frameworks = 'XCTest'` I wouldn't want to take that into consideration when deciding whether the xcconfig for the library spec should apply this fix or not. Using `:frameworks` from build settings includes that and we are only interested if the library itself brings in the `XCTest` framework

Reply:
Right, but the `frameworks` method on build settings is already scoped to the library specs when `library_xcconfig?` is true

Reply:
I am not sure that is entirely true for dynamic framework targets https://github.com/CocoaPods/CocoaPods/blob/master/lib/cocoapods/target/build_settings.rb#L570

This seems to be adding all consumer frameworks from all specs for dynamic framework targets.

Reply:
oooooh and this is what I was missing:
```ruby
    define_build_settings_method :spec_consumers, :memoized => true do
          if non_library_xcconfig?
            target.spec_consumers.select { |sc| non_library_spec == sc.spec }
          else
            target.spec_consumers.select { |sc| sc.spec.spec_type == @xcconfig_spec_type }
          end
        end
```
You are correct. I am going to update it.